src/prng/xorshift.rs

```diff
@@ -136,7 +136,8 @@ mod tests {
         assert_eq!(rng1.next_u64(), 4325440999699518727);
 
         let mut rng2 = XorShiftRng::from_rng(&mut rng1).unwrap();
-        assert_eq!(rng2.next_u64(), 15614385950550801700);
+        // Note: we cannot test the state of rng2 because from_rng does not
+        // fix Endianness. This is allowed in the trait specification.
```

Comment:
I was a little surprised that this change is only necessary for Xorshift. With HC-128 the test is (somehow) missing? And ChaCha passes because `from_rng` has a default implementation with `from_seed`, which does the endian swapping.

Does it make sense to still do the comparison, but make it dependent on the architecture?
(also, we now have an unused variable warning)

Reply:
My view is that if the documentation says `from_rng` isn't required to be consistent across platforms, there's no point testing consistency. Which means the ChaCha test is also kind-of wrong, but it doesn't hurt (currently).

Yeah, that explains the warning.

Reply:
I thought of the test as a way to also catch changes in the implementation. But don't care strongly.